### PR TITLE
Postgres db helm charts

### PR DESCRIPTION
**Pull request by gbuenodevsuse**
Adding postgres helm charts.

Reorganizing helm charts directories.

### CODE REVIEW

File: charts/postgres/templates/initjob.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    spec:
+      containers:
+        - name: init-db
+          image: postgres:16
+          command:
+            - sh
+            - -c
+            - |
+              until pg_isready -h postgres -U postgres; do sleep 1; done
+              psql -h postgres -U postgres -f /scripts/init.sql
```
Comment (rtamalin) [inline charts/postgres/templates/initjob.yaml:17]: Potentially we could use something like https://stackoverflow.com/a/17757560/8629780 to check for the existence of the DBs, skipping the init.sql if they already exist?

Or perhaps better we could follow the "Workaround from with psql" (first approach) in the accepted answer in https://stackoverflow.com/questions/18389124/simulate-create-database-if-not-exists-for-postgresql.

That would make the init.sql idempotent ;-)

File: charts/postgres/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}        
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
```
Comment (rtamalin) [inline charts/postgres/templates/deployment.yaml:36]: I know that `{{ .Values.image.tag | default .Chart.AppVersion }}` is the standard recommended substitution pattern, so it is reasonable to use.

However we are unlikely to be using the same versioning as the Postgres container for our chart so we should note in the corresponding values.yaml section that an appropriate tag value must be provided.

File: charts/postgres/values.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
+replicaCount: 1
+
+image:
+  repository: postgres
+  pullPolicy: IfNotPresent
+  tag: "16"
```
Comment (rtamalin) [inline charts/postgres/values.yaml:6]: We should note here that a valid tag must be specified that corresponds to the desired Postgres version we want to use.

**Review by rtamalin — APPROVED** [review]
Looks good - made some suggestions that do not need to be addressed in this PR; they can be done in follow up PRs.

File: charts/postgres/values.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+imagePullSecrets: []
+nameOverride: ""
+fullnameOverride: ""
+
+serviceAccount:
```
[the remark this answers is not in the document]
Comment (gbuenodevsuse) [reply charts/postgres/values.yaml]: I removed the defaults from helm create, which will not be used.

File: charts/postgres/values.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+affinity: {}
+
+autoscaling:
+  enabled: false
+  minReplicas: 1
+  maxReplicas: 3
+  targetCPUUtilizationPercentage: 80
+  # targetMemoryUtilizationPercentage: 80
+
+# OpenPlatform
```
[the remark this answers is not in the document]
Comment (rtamalin) [reply charts/postgres/values.yaml:45]: Note also that we will potentially have all 3 DB envs served by this one postgres instance, i.e staging (caching), operational and telemetry

**Comment by rtamalin** [comment]
LGTM - we can address any other changes in follow up PRs

File: charts/postgres/values.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (gbuenodevsuse) [reply charts/postgres/values.yaml:45]: That's a good suggestion. I'll have it addressed in a follow-up PR, together with k3s local dev environment setup.

**Review by rtamalin — APPROVED** [review]
LGTM

**Pull request merged**